queue_job/jobrunner/runner.py

```diff
@@ -332,7 +332,7 @@ def _query_requeue_dead_jobs(self):
                 state=(
                     CASE
                         WHEN
-                            max_retries IS NOT NULL AND
+                            NULLIF(max_retries, 0) IS NOT NULL AND
```

Comment:
```suggestion
                            max_retries IS NOT NULL AND
                            max_retries != 0 AND -- infinite retries if max_tries is 0
```

I would find this slightly more easy to understand but otherwise LGTM.

Reply:
Explicit is better than implicit.
Simple is better than complex.